src/components/ModalManager/ModalManager.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Transition, animated } from 'react-spring'
+import { breakpoint } from '@aragon/ui'
+
+const ModalContext = React.createContext({
+  component: null,
+  props: {},
+  showModal: () => {},
+  hideModal: () => {},
+})
+
+class ModalProvider extends React.Component {
+  showModal = (component, props = {}) => {
+    this.setState({ component, props })
+  }
+
+  hideModal = () => {
+    this.setState({ component: null, props: {} })
+  }
+
+  state = {
+    component: null,
+    props: {},
+    showModal: this.showModal,
+    hideModal: this.hideModal,
+  }
+
+  render() {
+    return (
+      <ModalContext.Provider value={this.state}>
+        {this.props.children}
+      </ModalContext.Provider>
+    )
+  }
+}
+
+const ModalConsumer = ModalContext.Consumer
+
+const ModalView = () => (
+  <ModalConsumer>
+    {({ component: Component, props, hideModal }) => (
+      <Transition
+        native
+        from={{ opacity: 0, top: 50 }}
+        enter={{ opacity: 1, top: 0 }}
+        leave={{ opacity: 0, top: 0 }}
```

Comment:
Something that's not nice when doing this leave animation is that the user's blocked from being able to interact with the screen until the transition's completely done. I think @bpierre has encountered this before and solved it (maybe with the side panels?)... but not sure.

Reply:
[`pointer-events: none`](https://developer.mozilla.org/en-US/docs/Web/CSS/pointer-events) 😆

Reply:
`pointer-events` AND `hack-tricks` from @bpierre with `react-spring`

Reply:
Can confirm this is working! Awesome guys! 💫